registry/session.go

```diff
@@ -303,8 +310,19 @@ func (r *Session) GetRemoteTags(registries []string, repository string) (map[str
 		}
 
 		result := make(map[string]string)
-		if err := json.NewDecoder(res.Body).Decode(&result); err != nil {
-			return nil, err
+		if askedTag == "" {
```

Comment:
Please add a separate method for `GetRemoteTags` and `GetRemoteTag`. These have fundamentally different behaviors and there is no reason to have shared functions.